test/unit/pipeline_test.rb

```diff
@@ -82,5 +82,70 @@ class PipelineTest < ActiveSupport::TestCase
         assert @pipeline2.is_read_length_consistent_for_batch?(@batch)
       end
     end
+
+    context 'sequencing_pipeline#requested_flowcell_type consistency among batch requests' do
+      setup do
+        @sample = create :sample
+
+        @request_type = create :request_type, name: 'sequencing', target_asset_type: nil
+        @pipeline = create :sequencing_pipeline, name: 'sequencing pipeline', request_types: [@request_type]
+        @request1 =
+          create(
+            :sequencing_request,
+            asset: create(:sample_tube, :scanned_into_lab, sample: @sample),
+            target_asset: nil,
+            request_type: @request_type
+          )
+
+        @request2 =
+          create(
+            :sequencing_request,
+            asset: create(:sample_tube, :scanned_into_lab, sample: @sample),
+            target_asset: nil,
+            request_type: @request_type
+          )
+      end
+
+      should 'return true if no request was selected' do
+        @batch = create :batch
+        assert @pipeline.is_flowcell_type_consistent_for_batch?(@batch)
+      end
+
+      should "return true if the requests don't make use of the requested_flowcell_type attribute" do
+        @batch = @pipeline.batches.create!(requests: [@request1, @request2])
+        assert @pipeline.is_flowcell_type_consistent_for_batch?(@batch)
+      end
+
+      should 'check that the requested_flowcell_type attribute is the same in all the requests' do
+        @request1.request_metadata.requested_flowcell_type = 'S2'
+        @request2.request_metadata.requested_flowcell_type = 'S4'
+        @batch = @pipeline.batches.create(requests: [@request1, @request2])
+        assert_not @pipeline.is_flowcell_type_consistent_for_batch?(@batch)
+      end
+
+      should 'check that other pipelines are not affected by different requested_flowcell_type attributes' do
+        @pipeline2 = create :pipeline, name: 'other pipeline', request_types: [@request_type]
+        @request1 =
+          create(
+            :sequencing_request,
+            asset: create(:sample_tube, sample: @sample),
+            target_asset: nil,
+            request_type: @request_type
+          )
+
+        @request2 =
+          create(
+            :sequencing_request,
+            asset: create(:sample_tube, sample: @sample),
+            target_asset: nil,
+            request_type: @request_type
+          )
+
+        @request1.request_metadata.requested_flowcell_type = 'S2'
+        @request2.request_metadata.requested_flowcell_type = 'S4'
+        @batch = @pipeline2.batches.create(requests: [@request1, @request2])
+        assert @pipeline2.is_flowcell_type_consistent_for_batch?(@batch)
```

Comment:
Why does it pass for pipeline2?

Reply:
The validation check is applied only to the sequencing_pipeline. Other pipelines are not validated.